### PR TITLE
feat: add new challenge 45 React in Angular

### PR DESCRIPTION
**Pull request by wandri**
Add new challenge: React in Angular.

The goal of this challenge is to use a React component inside an Angular application.

Many components are available in React, and it can be interesting to use them in an Angular application. The goal is to create a React component and use it in an Angular application.

## Information

In this challenge, we have a simple application and a react component `ReactPost` in `app/react` to illustrate a react component from a library.

## Statement

- Your task is to display the posts with the React component `ReactPost`.
- When you select a post, the post should be highlighted.

In order to play with the react component, you should start by installing the react dependencies.

```bash
npm i --save react react-dom
npm i --save-dev @types/react @types/react-dom
```

## Constraints

- Do not transform the react component in an angular component. The React component is pretty simple and can be written with ease in Angular. But **the goal is to use the React component**.

### Hint

<details>
  <summary>Hint 1 - Configuration</summary>
  Allow the React files in tsconfig.json

```
{
...
"compilerOptions": {
  ...
  "jsx": "react"
},
...
}
```

</details>

<details>
  <summary>Hint 2 - Initialization</summary>
  Create a react root with `createRoot(...)`
</details>

<details>
  <summary>Hint 3 - Display</summary>
  To render the component, it should look like this:
  
    ```
    <react root>.render(
        <React.StrictMode>
        ...
        </React.StrictMode>
    )
    ``` 
</details>

<details>
  <summary>Hint 4 - Design</summary>
  Do not forget to allow the react file in Tailwind.
</details>

demo-question-45-react-in-angular

### CODE REVIEW

**Comment by tomalaforge** [comment]
Hey, that's a really nice challenge. I will try it myself 💪 
I have no idea how to do that.

However can you delete the react dependancies from the main branch. I will try to delete packages that are not used very often and include them only in the branch that need them. And since you explained it inside the documentation. that will not be a problem. 

When this is done, i will merge your challenge. 

thanks a lot 🙏

File: apps/angular/react-in-angular/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# React in angular
+
+> author: wandrille-guesdon
```
Comment (tomalaforge) [inline apps/angular/react-in-angular/README.md:3]: you need to create a file inside the docs/src/content/authors otherwise the application will not build

**Comment by wandri** [comment]
@tomalaforge Thanks for the feedback. I will revert the solution (which contains the react package)

File: apps/angular/react-in-angular/README.md (hunk 1 above)
Comment (wandri) [reply apps/angular/react-in-angular/README.md:3]: Thanks, ti's done

**Comment by wandri** [comment]
fix: the solution has been reverted.
fix; the author file has been added.

File: docs/src/content/docs/challenges/angular/45-react-in-angular.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+---
+title: 🔴 React in angular
+description: Challenge 45 is about learning how to benefit from the numerous libraries in React
+author: Wandrille Guesdon
```
Comment (tomalaforge) [inline docs/src/content/docs/challenges/angular/45-react-in-angular.md:4]: the author must match the name of the file

**Comment by tomalaforge** [comment]
Once the challenge is merged, will you be able to provide a solution next week or the week after ?

File: docs/src/content/docs/challenges/angular/45-react-in-angular.md (hunk 2 above)
Comment (wandri) [reply docs/src/content/docs/challenges/angular/45-react-in-angular.md]: Ah, well noticed! thanks. It's fixed

**Comment by wandri** [comment]
The solution is already Ready. I need first to be able to rebase when it's merged.

**Review by tomalaforge — APPROVED** [review]

**Comment by tomalaforge** [comment]
@all-contributors add @wandri for challenge

**Comment by tomalaforge** [comment]
@all-contributors add @wandri for code

**Pull request merged**